### PR TITLE
RUST-2030 Add more event fields: lsid, txnNumber and disambiguatedPaths

### PR DESCRIPTION
**Pull request by arthurprs**
Add more event fields: lsid, txnNumber and disambiguatedPaths

### CODE REVIEW

File: src/test/change_stream.rs [hunk 1]
```diff
@@ -659,6 +659,48 @@ async fn split_large_event() -> Result<()> {
...
+
+    let next_event = stream.next().await.transpose()?;
+    assert!(matches!(next_event,
+        Some(ChangeStreamEvent {
+            operation_type: OperationType::Insert,
+            document_key: Some(key),
+            lsid: Some(lsid),
+            txn_number: Some(1),
+            ..
+        }) if key == doc! { "_id": 1 } && lsid.get("id") == session_id.as_ref()
```
Comment (arthurprs) [inline src/test/change_stream.rs:698]: The lsid contains a second field "uid", which I'm familiar with. But the test should still be valid.

**Comment by isabelatkinson** [comment]
Hey @arthurprs, thanks for opening this PR! We have an open ticket for adding the `disambiguated_paths` field (RUST-1423) which also includes an option and some additional testing, so we can certainly address that.

What is your use case for adding the `txn_number` and `lsid` fields? These are not part of the specification for the `ChangeStreamEvent` type, so I'm hesitant to make this addition. Would it be possible for you to retrieve these values from [`CommandStartedEvent`](https://docs.rs/mongodb/latest/mongodb/event/command/struct.CommandStartedEvent.html)s emitted via command monitoring/logging instead? They would both be present as top-level fields in the [`document`](https://docs.rs/mongodb/latest/mongodb/event/command/struct.CommandStartedEvent.html#structfield.command) field for events emitted during a transaction.

**Comment by arthurprs** [comment]
Maybe I'm misinterpreting [the manual page](https://www.mongodb.com/docs/manual/reference/change-events/update/#mongodb-data-update) but `lsid` and `txn_number` seemed documented and even referred to by `clusterTime` description.

Our use cases include detecting if a certain actor was the source of the write by its session ID (or previous ones).

**Comment by abr-egn** [comment]
You're quite right, those fields are in the manual!  The confusing part here on our end is that they're not part of the [specification](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.md#server-specification), so I'm working to track down that discrepancy.  My concern is that I don't want to add fields to the public API if they're not actually intended to be part of the stable interface 🙂 

While we're tracking this down, you should be able to unblock anything that needs those fields by defining your own event type and using it with [`ChangeStream::with_type`](https://docs.rs/mongodb/latest/mongodb/change_stream/struct.ChangeStream.html#method.with_type).

**Review by abr-egn — APPROVED** [review]
LGTM!

**Comment by abr-egn** [comment]
Sorry for the delay here, it looks like it was just a simple oversight that those fields weren't included in the spec.  I've started the test run and assuming there are no unexpected failures I'll merge this in.  Thanks again for your patience!

**Comment by abr-egn** [comment]
It looks like the test is using some logic that's not supported in server version 4.x, can you add a check to restrict it to >=5.0?

**Comment by arthurprs** [comment]
Done in 
bba9d00

**Pull request merged**